lib/pipefy_message/publisher.rb

```diff
@@ -14,8 +14,12 @@ def initialize(broker = "aws", broker_opts = {})
       @publisher_instance = build_publisher_instance
     end
 
-    def publish(message, topic, context = nil)
-      @publisher_instance.publish(message, topic, context)
+    def publish(message, topic, context = nil, correlation_id = nil)
+      context = "NO_CONTEXT_PROVIDED" if context.nil?
+      correlation_id = "NO_CID_PROVIDED" if correlation_id.nil?
+      event_id = SecureRandom.uuid.to_s
```

Comment:
**question:** I used this method because the previous version of the gem used it to generate the correlation ID inside the SNS broker, but if there are any better strategies please add your suggestions.

Reply:
I believe this is safe enough. Very very (_very!_) unlikely to be collision prone.